schema/server/branch.json

```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "version": "0.0.1",
+  "type": "object",
+  "properties": {
+    "name": {
+      "oneOf": [
+        { "type": "null" },
+        { "type": "string" }
+      ]
+    },
+    "commit_hash": {
+      "type": "string"
+    },
+    "time": {
+      "type": "number"
+    },
+    "message": {
```

Comment:
If you follow the suggestion above in the README this would become an array of strings (which might be zero length in the case of an empty message)

Reply:
okay ill still keep the null value to indicate message was not in utf-8 distinct from an empty message

Reply:
You could also do a lossy conversion if you hit bad utf8 characters: https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy